src/units/abilities.cpp

```diff
@@ -899,16 +900,20 @@ std::string attack_type::weapon_specials(bool only_active, bool is_backstab) con
 			for (const config::any_child sp : it->abilities().all_children_range()){
 				const bool active = check_adj_abilities_impl(shared_from_this(), other_attack_, sp.cfg, self_, *it, i, self_loc_, AFFECT_SELF, sp.key);
 
-				add_name(weapon_abilities, active, sp, checking_name);
+				add_name(weapon_abilities, active, sp, checking_name, gender);
 			}
 		}
 	}
 
 	if(other_){
+		unit_race::GENDER gender = other_->gender();
+		if(self_){
```

Comment:
Is this check even correct? You checked the "have a self" case just above here… but if you have a self _and_ an other, this means it uses the gender of the self rather than that of the other. Is that correct though?

Reply:
Correct,self_ is the unit who own attack affected by abilitie, and 'special_efffect' describe the effect on attack.

this is the attack of self who has 'decreased presision', in attack prediction windows and not the unit itself, this is why idon't understand the idea of female variant because for me; attack is a female name and never male (in french)

Reply:
I'm not 100% sure that it makes sense to have male and female strings here; however, it should be consistent. Either both name and special effects have male/female versions, or neither do.

Reply:
i remove special_effect attribute